andela/denethor-ah-backend
# #163478793 Implement User Registration

#### What does this PR do?
- Implement User registration
- implement authentication using Passport-JWT strategy
- implement input validation using Joi 
- implement endpoint to register user 
- implement endpoint to verify user 
- implement email client (Sendgrid API)

#### How should this be manually tested?
- clone the repo and checkout to this feature branch
- run `npm install`
- setup database and run `npm run migrate`
- run `npm run start:dev` to start applicarion
- make a `POST` request to the `/api/users/`  to register with valid credentials
- make a `PATCH` request to the `/api/users/:token` to verify registration
- check email for verification mail
 
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/163478793
https://www.pivotaltracker.com/story/show/163478794
https://www.pivotaltracker.com/story/show/163478795
